### DISCUSSION TITLE
Tooling a Book

### DISCUSSION THREAD

**Discussion by JacquesCarette** [JacquesCarette/Drasil#3555] [Ideas]
I found the blog post [Tooling a Book](https://third-bit.com/2023/07/16/tooling-a-book/) very interesting as a source of (potential) requirements for Drasil. He's got an approach to things that is highly compatible with "generate everything" thinking.

**Comment by smiths** [comment]
Yes, I can see why this caught your attention.  I read a bunch of it thinking how much it sounded like Greg Wilson, then I clicked on the About Me link and learned that it is written by Greg Wilson.  :-)  I haven't looked at all the details, but it feels more like sophisticated literate programming, than it does to "generate everything."  I couldn't find a link to the source code for the book - the code that was the input for the generator.  Is that online somewhere?

**Reply by JacquesCarette** [reply]
It is not "generate everything" but it sure does generate a lot more stuff than mere literate programming. [I didn't look for the source.]

The features that he implements seemed like a good source of ideas for features that Drasil should have too.
